src/screens/NotePropertiesScreen.tsx

```diff
@@ -0,0 +1,217 @@
+// SPDX-FileCopyrightText: © 2019 EteSync Authors
+// SPDX-License-Identifier: GPL-3.0-only
+
+import * as React from "react";
+import { useSelector } from "react-redux";
+import { View } from "react-native";
+import { Text, HelperText, Button, TouchableRipple } from "react-native-paper";
+import { useNavigation, RouteProp } from "@react-navigation/native";
+import { StackNavigationProp } from "@react-navigation/stack";
+import { Item } from "etebase";
+
+import { useSyncGate } from "../SyncGate";
+import { useCredentials } from "../credentials";
+import { StoreState, useAsyncDispatch } from "../store";
+import { itemBatch, pushMessage } from "../store/actions";
+import { CachedItem } from "../store/reducers";
+
+import TextInput from "../widgets/TextInput";
+import ScrollView from "../widgets/ScrollView";
+import Container from "../widgets/Container";
+import ErrorOrLoadingDialog from "../widgets/ErrorOrLoadingDialog";
+import Select from "../widgets/Select";
+import TextInputWithIcon from "../widgets/TextInputWithIcon";
+
+import { useLoading, navigateTo404, NoteMetadata } from "../helpers";
+import { RootStackParamList } from "../RootStackParamList";
+
+interface FormErrors {
+  name?: string;
+  notebook?: string;
+}
+
+type NavigationProp = StackNavigationProp<RootStackParamList, "NoteProps"> | StackNavigationProp<RootStackParamList, "NoteCreate">;
+
+interface PropsType {
+  route: RouteProp<RootStackParamList, "NoteProps"> | RouteProp<RootStackParamList, "NoteCreate">;
+}
+
+export default function NotePropertiesScreen(props: PropsType) {
+  const colUid = props.route.params?.colUid ?? "";
+  const itemUid = (props.route.params && "itemUid" in props.route.params)
```

Comment:
This should really probably be: `props.route.params?.itemUid ?? ""` if that's what you are trying to achieve, though see my other comment about maybe not using "" but instead use `null`.

Reply:
I had to do that because of the RootStackParamList type.

I created two URLs for web:

```javascript
NoteCreate: "notes/new-note",
NoteProps: "notes/:colUid/:itemUid/properties",
```
The first one can receive `colUid` so it's preselected in the Notebooks select, and the second one receives of course `colUid` and `itemUid`. So I have those types in RootStackParamList:

```javascript
NoteCreate: { colUid: string } | undefined;
NoteProps: {
  colUid: string;
  itemUid: string;
};
```
And the params type is defined like that:
```javascript
params?: Readonly<{
    colUid: string;
    itemUid: string;
}> | Readonly<{
    colUid: string;
}> | undefined
```
If I put your line instead of mine, I then get the warning `Property 'itemUid' does not exist on type 'Readonly<{ colUid: string; }>'.`. And this is needed even with `const itemUid = props.route.params?.itemUid`

I could put the same types for both, but then that wouldn't be properly checked in `navigate`.

Reply:
Ah, I thought the type checking handled this case still, I guess I was wrong.

Reply:
Actually a solution could be to define the NoteCreate params like that:

```javascript
NoteCreate: {
    colUid: string;
    itemUid?: undefined;
} | undefined;
```

Reply:
If not create could accept itemUid then add it, if it can't, then don't add it. It's better to need to use the verbose syntax you used rather than have wrong typing.

Reply:
There will be proper typing, itemUid will have to be set as undefined if it is set as a param for NoteCreate, which is what we expect anyway.

Reply:
Great, fixed?